clients/src/test/java/org/apache/kafka/common/requests/ProduceResponseTest.java

```diff
@@ -100,16 +113,92 @@ public void produceResponseRecordErrorsTest() {
             ProduceResponse response = new ProduceResponse(responseData);
             Struct struct = response.toStruct(ver);
             assertEquals("Should use schema version " + ver, ApiKeys.PRODUCE.responseSchema(ver), struct.schema());
-            ProduceResponse.PartitionResponse deserialized = new ProduceResponse(struct).responses().get(tp);
+            ProduceResponse.PartitionResponse deserialized = new ProduceResponse(new ProduceResponseData(struct, ver)).responses().get(tp);
             if (ver >= 8) {
                 assertEquals(1, deserialized.recordErrors.size());
                 assertEquals(3, deserialized.recordErrors.get(0).batchIndex);
                 assertEquals("Record error", deserialized.recordErrors.get(0).message);
                 assertEquals("Produce failed", deserialized.errorMessage);
             } else {
                 assertEquals(0, deserialized.recordErrors.size());
-                assertEquals(null, deserialized.errorMessage);
+                assertNull(deserialized.errorMessage);
             }
         }
     }
+
+    /**
+     * the schema in this test is from previous code and the automatic protocol should be compatible to previous schema.
+     */
+    @Test
+    public void testCompatibility() {
```

Comment:
I think this test might be overkill. We haven't done anything like this for the other converted APIs. It's a little similar to `MessageTest.testRequestSchemas`, which was useful verifying the generated schemas when the message generator was being written. Soon `testRequestSchemas` will be redundant, so I guess we have to decide if we just trust the generator and our compatibility system tests or if we want some other canonical representation. Thoughts?

Reply:
That makes sense to me. Will remove redundant test.